### PR TITLE
JBPM-5861: Added support for PDF export settings.

### PR DESCRIPTION
**Pull request by romartin**
Hey @ederign @porcelli @manstis 

Added support for custom settings for the PDF export feature. This way different orientation and sizes can be specified too. 

This commit basically does:
- Refactored class name from `PdfContent` to `PdfDocument`, as I think it makes more sense now that it contains the document settings too, not just the content. The commit that added `PdfContent` has been merged a few hours ago, so do not expect big issues at this point due to this, hope is good for you too :)
- Added `PdfSettings`, the new native jstype and added it as optional argument in the factory methods.

Thanks!

### CODE REVIEW

**Comment by romartin** [comment]
ok to test

**Comment by romartin** [comment]
hey @paulovmr @porcelli @ederign 
Notice using the Preferences API for file export settings will make the `uberfire-commons-editor` modules depend on the Preferences ones (actually it doesn't). So from that point, all webapps will have to include the preferences module dependencies, otherwise the build will fail. Does that makes sense for you? Are you expecting or already know if something will get broken doing this? 
Thanks!

**Comment by paulovmr** [comment]
Hi @romartin, I don't see any problem on this, tbh I think all webapps already depend on it, at least most of them.

**Comment by romartin** [comment]
Hey @porcelli @manstis @ederign @paulovmr 
Updated this commit by the use of the Preferences API - added file export settings as workbench preferences. 
Thanks!

File: uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-api/src/main/resources/META-INF/beans.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<beans xmlns="http://java.sun.com/xml/ns/javaee"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="
```
Comment (manstis) [inline uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-api/src/main/resources/META-INF/beans.xml]: Nit-pick about the schema location being partially on a different line..

Comment (romartin) [inline uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-api/src/main/resources/META-INF/beans.xml]: oh well seen @manstis thansk! In fact, I copy-pasted it from another module in uberfire, so did the review and fixed 2 like these found. Thanks!

**Comment by romartin** [comment]
Added a new commit for fixing @manstis comments. Thanks!

**Comment by porcelli** [comment]
jenkins please test this

**Comment by manstis** [comment]
@romartin IMO this has enough approval to merge; but was you wanting @paulovmr and @ederign to approve?

**Comment by ederign** [comment]
thanks @romartin and sorry for the delay.

**Pull request merged**